src/plonk/circuit.rs

```diff
@@ -460,14 +460,51 @@ impl<F> Mul<F> for Expression<F> {
 #[derive(Copy, Clone, Debug)]
 pub(crate) struct PointIndex(pub usize);
 
+/// A "virtual register" is a PLONK cell that has been queried at a particular relative
+/// offset within a custom gate.
```

Comment:
Why not call it a register? It's no more or less virtual than anything else 😝

Reply:
Or how about `VirtualCell`? A cell that has yet to be allocated an absolute position.

Reply:
I'll go with `VirtualCell` for now, because I'd rather not use "register" on its own yet while I haven't figured out the downstream parts of the API change (currently I have concepts of "virtual register" and "concrete register" that I want to separate, but this might change).